compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -754,6 +754,13 @@ class ResolverImpl(
                 )
             }
             val typeSubstitutor = containing.kotlinType.createTypeSubstitutor()
+            if (declaration is PropertyAccessorDescriptor) {
+                val substitutedProperty = (declaration.correspondingProperty).substitute(typeSubstitutor)
+                // TODO: Avoid relying on naming convention.
+                // Fix in upstream for property accessors: https://github.com/JetBrains/kotlin/blob/master/core/descriptors/src/org/jetbrains/kotlin/descriptors/impl/PropertyAccessorDescriptorImpl.java#L122
+                val prefix = function.simpleName.asString().substring(0,3)
+                    return KSFunctionImpl((substitutedProperty as PropertyDescriptor).accessors.single { it.name.asString().substring(1,4) == prefix })
```

Comment:
Would `it.name == declaration.name` make sense? So that you don't rely on the naming convention.